backend/spiritual_avatar_generation_engine.py

```
@@ -65,296 +67,144 @@ async def generate_complete_avatar_video(
         start_time = time.time()
-        
-        while time.time() - start_time < max_wait:
+        while time.time() - start_time < timeout:
             try:
+                await asyncio.sleep(10)
                 async with aiohttp.ClientSession() as session:
                     async with session.get(
                         f"{self.d_id_base_url}/talks/{talk_id}",
-                        headers=headers
+                        headers={"Authorization": f"Basic {self.settings.d_id_api_key}"}
                     ) as response:
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)